src/index.tsx

```diff
@@ -11,7 +11,7 @@ import { store }       from './store';
 
 ReactDOM.render(
     <Provider store={store}>
-        <Node key={1} nodeID={1} />
+        <Node key={2} nodeID={2} />
```

Comment:
Without any further investigation my gut feeling is that records that do not conform to INode should not be in the database.